Merging develop into master to add os2forms_forloeb_profile and Docksal environment update

madsnorgaard opened the pull request:
This PR includes following changes:

- The addition of a new install profile for OS2forms med Forløb
- Updates of Docksal environment
- Additional documentation for development environment and deployment
- Inclusion of os2forms_forloeb module in composer.json
- A variety of changes to themes from Bellcom additions to the develop branch
- All Maestro and webform templates from 3rd sprint

Test procedure with Docksal:

1. Clone the git repository and checkout this PR
   ```sh
   git clone git@github.com:OS2Forms/os2forms8.git
   ```

2. Enter the newly created project directory
   ```sh
   cd os2forms8
   ```

3. Start docksal environment
   ```sh
   fin start
   ```

4. Create local settings
   ```sh
   cp web/sites/example.settings.local.php web/sites/default/settings.local.php
   ```

5. Add file permission fix to settings.local.php. See https://docs.docksal.io/apps/drupal/#file-permissions
   ```php
   // web/sites/default/settings.local.php

   $settings['file_chmod_directory'] = 0777;
   $settings['file_chmod_file'] = 0666;
   ```

6. Install dependencies and enable modules
   ```sh
   fin build-forloeb
   ```

7. Configure trusted hosts in settings.local.php (add the following if not present)
   ```php
   // web/sites/default/settings.local.php

   $settings['trusted_host_patterns'] = ['^os2forms8.docksal$', '^localhost$'];
   ```
You should now be able to browse to the application at `http://os2forms8.docksal`

lats1 commented on composer.lock:
```diff
@@ -9992,18 +11529,99 @@
+            "source": {
+                "type": "git",
+                "url": "https://github.com/OS2Forms/os2forms_forloeb.git",
+                "reference": "9c6c9acbd982a723ecf836637d4b4a600d0b82f1"
```
This points to a commit on the develop branch in the os2forms_forloeb repository, that was merged in 28 days ago. This means that if this is merged into the master branch, and you pull the master branch, and run a composer install --no-dev, the version of the installed os2forms_forloeb package will be that commit from 28 days ago...

This is all good if the master branch of os2forms8 should include that version of the os2forms_forloeb package. Is that the case?

Another question. Shouldn't the production ready code of os2forms8 include the production ready code of the os2forms_forloeb package?

agger-magenta replied:
Good points. We should probably tag the production ready version of `os2forms_forloeb` that we wish to install and fetch it by that tag in order to remove doubts.

madsnorgaard replied:
I have updated the composer-lock file and it is up to date.

andriyun commented on composer.json:
```diff
@@ -35,14 +35,15 @@
         "os2core/os2core": "^1.0.3",
         "os2forms/os2forms": "^2.5",
         "os2forms/os2forms_egbolig": "dev-master",
+        "os2forms/os2forms_forloeb": "dev-develop",
```
Hej @madsnorgaard 
One of demand on meeting was to avoid dev references from packagist.
If os2forms/os2forms_forloeb is ready to merge into master I think it should get fixed git tag. Isn't it?

madsnorgaard replied:
Hi @andriyun, you are right - we should add tags for versioning at this point.

andriyun commented:
Hi mates
Just want to inform you I've added some changes not related to os2forms_forloeb.
We are getting requests for project extending/improving.
In my last commit I've added to contrib modules that @asl-hk have requested.

madsnorgaard commented:
> Hi mates
> Just want to inform you that I've added some changes not related to os2forms_forloeb.
> We are getting requests for project extending/improving.
> In my last commit I've added to contrib modules that @asl-hk have requested.

Straw module is a welcome addition. Is this in regards to the import of data via REST to taxonomies in OS2forms?

andriyun commented:
I have no info about it. So I din't know the final use case.

andriyun commented on web/profiles/custom/os2forms_forloeb_profile/os2forms_forloeb_profile.install:
```diff
@@ -0,0 +1,56 @@
+ *
+ * Enabled
+ */
+function os2forms8_install_tasks(&$install_state) {
```
This hook will not be run such as it has not correct name

andriyun commented on web/profiles/custom/os2forms_forloeb_profile/os2forms_forloeb_profile.install:
```diff
@@ -0,0 +1,56 @@
+
+/**
+ * @file
+ * Install, update and uninstall functions for the os2forms8 installation profile.
```
Fix profile name

andriyun commented on web/profiles/custom/os2forms_forloeb_profile/os2forms_forloeb_profile.install:
```diff
@@ -0,0 +1,56 @@
+ *
+ * @see os2forms8_install_tasks()
+ */
+function os2forms8_install_simplesaml_roles() {
```
This function should be renamed with profile name suffix too

andriyun reviewed (dismissed):
Hi @madsnorgaard 

I found few tiny things that I should be fixed before we merge it.

And I run upgrade status check on `os2forms_forloeb` module and found deprecated function usage in line
https://github.com/OS2Forms/os2forms_forloeb/blob/ce92abc35dc0472294f6105edf8494a417467f71/os2forms_forloeb.module#L97

```
FILE: web/modules/contrib/os2forms_forloeb/os2forms_forloeb.module

STATUS         LINE                           MESSAGE                           
--------------------------------------------------------------------------------
Fix now        97   Call to deprecated function drupal_set_message(). Deprecated
                    in drupal:8.5.0 and is removed from drupal:9.0.0.  Use      
                    Drupal\Core\Messenger\MessengerInterface::addMessage()      
                    instead.   
```

andriyun commented:
I also added two issues in https://github.com/OS2Forms/os2forms_forloeb/issues repository

andriyun commented:
Shouldn't reference to os2forms_forloeb package be updated then to the new version that include fixes?

madsnorgaard commented:
@andriyun updated version 1.1.0 has been added to composer.json and composer.lock

madsnorgaard commented:
We should be ready for merge. I am ready to make any necessary changes if needed.

andriyun commented:
LGTM :)
